### DISCUSSION TITLE
Run launcher inside home network

### DISCUSSION THREAD

**Discussion by MaciekMr** [Burralis/Game-Launcher#49] [Q&A]
Hi, I have run the wow server (trinity) on my local network (linux machine) with ip 11.11.1.40 and trying to connect from windows machine from ip 11.11.1.29. 
Even run launcher with option: "Arctium WoW Launcher.exe" --dev
it rejected the connection:
_____________World of Warcraft___________
                   _   _
    /\            | | (_)
   /  \   _ __ ___| |_ _ _   _ _ __ ___
  / /\ \ | '__/ __| __| | | | | '_ ` _ \
 / ____ \| | | (__| |_| | |_| | | | | | |
/_/    \_\_|  \___|\__|_|\__,_|_| |_| |_|

___________WoW Client Launcher___________
            https://arctium.io

Operating System: Microsoft Windows 10.0.22631
Mode: Custom Server (Retail)

Developer mode: Disabled

Client Portal '11.11.1.40 '
Server with host name 11.11.1.40  does not have a trusted certificate attached.
If you are the server owner be sure to generate one and replace the default bnet server certificate.
One way to generate one is through Let's Encrypt.
Closing in 5 seconds...

I cant generate certificates, as it is not a public server, just local one, using local network. Any advice how to proceed?

best regards

**Comment by ZaDarkSide** [comment]
These are the only allowed IP CIDR ranges allowed in the software: https://github.com/Arctium/WoW-Launcher/blob/windows/src/Program.cs#L37-L59
Basically you are using for internal network internet IPs and that's wrong. 11.x.x.x is supposed to be used for internet only not LANs.

**Comment by MaciekMr** [comment]
I understand the range of ip 11.x.x.x is mostly reserved for internet, but I have always configure my local lan little bit differently. Is it possible to add custom ip range as --dev parameter?
btw, may I use the latest sources (git) and add my ip range localy and run it with 10.2.0.52649 Wow version?

**Reply by Fabi** [reply]
There is no support for custom ip ranges. Also no support as soon as you edit the source from our side. However you can change the source to add custom ranges.
BUT: It will crash with that client build and the source on the repo so you are on your own regarding that.

**Reply by MaciekMr** [reply]
I've read the repository (source code) will be closed (or it is already closed). I understand the solution for custom ip range will be added in the future :).
I saw there is a temporary solution for crash, but it is not implemented in opensource. Any advice what cause the crash to add it to source code localy ? :)

**Reply by Fabi** [reply]
There won't be any solution for custom ip ranges in future. There will also be no open source commits anymore including the temp crash fix for the current 10.2.0 build. So you are on your own on that, sorry
